src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -235,8 +235,8 @@ private BufferSegment AllocateSegment(int sizeHint)
             }
             else
             {
-                // We can't use the pool so allocate an array
-                newSegment.SetUnownedMemory(new byte[sizeHint]);
+                // We can't use the pool so use the array pool
```

Comment:
Nit: "the pool" => "the memory pool" (so as to distinguish it from "the array pool")?  (Same in other places.)

Reply:
Removed the comment as combined the two arraypool accesses